Microsoft.Recognizers.Text.DateTime/Parsers/BaseTimeParser.cs

```diff
@@ -183,22 +183,28 @@ private DateTimeResolutionResult Match2Time(Match match, DateObject referenceTim
             var descStr = match.Groups["desc"].Value.ToLower();
             if (!string.IsNullOrEmpty(descStr))
             {
-                if (descStr.ToLower().StartsWith("a"))
+                if (descStr.ToLower().StartsWith(config.UtilityConfiguration.AmPrefix))
                 {
                     if (hour >= 12)
                     {
                         hour -= 12;
                     }
                     hasAm = true;
                 }
-                else if (descStr.ToLower().StartsWith("p"))
+                else if (descStr.ToLower().StartsWith(config.UtilityConfiguration.PmPrefix))
                 {
                     if (hour < 12)
                     {
                         hour += 12;
                     }
                     hasPm = true;
                 }
+
+                //set the hasAm to false since it is ampm
+                if (descStr.ToLower().StartsWith(config.UtilityConfiguration.AmPmPrefix))
```

Comment:
This could be the first if in the if/else sequence. Or only check the other if if this is false. Either will make code cleaner.
Also, please add a comment to the code describing that "ampm" is a special token.